pkg/templatelookup/regular.go

```diff
@@ -52,12 +53,23 @@ func (t *TemplateLookup) GetRegularTemplates(ctx context.Context, kyma *v1beta2.
 		if found {
 			continue
 		}
-		if !module.Valid {
-			templates[module.Name] = &ModuleTemplateInfo{Err: fmt.Errorf("%w: invalid module", ErrTemplateNotValid)}
+		if !module.IsValid() {
+			if module.Enabled {
+				templates[module.Name] = &ModuleTemplateInfo{Err: fmt.Errorf("%w: %s", ErrModuleInSpecNotValid, module.ValidationError)}
```

Comment:
what's the purpose to distinguish those error in spec and in status?

Reply:
Well, only to provide more accurate error message so that we know what has caused the problem. I have found the existing "given module template is not valid": invalid module" error totally useless considering it was used to describe a problem with missing manifest reference in the Kyma.Status.

Reply:
Done: I am now returning an error directly, so this conditional could be removed.